packages/dashboard/src/components/DashboardFilters/DashboardFiltersEditor/components/FilterEditor.tsx

```diff
@@ -77,21 +80,55 @@ const FilterEditor: React.FC<FilterEditorProps> = ({ filter, config, updateFilte
       }
 
       if (_dataSet.data) {
-        _dataSet.data.forEach(row => {
-          Object.keys(row).forEach(columnName => {
-            columns[columnName] = true
-          })
-        })
+        callback(_dataSet.data)

```

Comment:
could we be more descriptive of what callback is doing?  something generic like "transformData" could be more descriptive